solutions/silver/usaco-669.mdx

```diff
@@ -315,11 +315,11 @@ int main() {
 		read >> c.first >> c.second;
 	}
 
-	int lo = 0;
-	int hi = 1e9;  // arbitrarily large integer
-	int valid = -1;
+	long long lo = 0;
+	long long hi = INT32_MAX;  // arbitrarily large integer
+	long long valid = -1;
 	while (lo <= hi) {
-		int mid = (lo + hi) / 2;
```

Comment:
you should only need to change this to `mid = lo + (hi - lo) / 2`. then everything can still be ints